x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/KibanaOwnedReservedRoleDescriptors.java

```diff
@@ -378,7 +379,13 @@ static RoleDescriptor kibanaSystem(String name) {
                 // Endpoint heartbeat. Kibana reads from these to determine metering/billing for endpoints.
                 RoleDescriptor.IndicesPrivileges.builder().indices(".logs-endpoint.heartbeat-*").privileges("read").build(),
                 // For connectors telemetry. Will be removed once we switched to connectors API
-                RoleDescriptor.IndicesPrivileges.builder().indices(".elastic-connectors*").privileges("read").build() },
+                RoleDescriptor.IndicesPrivileges.builder().indices(".elastic-connectors*").privileges("read").build(),
+                // Connector secrets. Kibana can only write to this index.
+                RoleDescriptor.IndicesPrivileges.builder()
+                    .indices(".connector-secrets*")
+                    .privileges("write", "delete", "create_index")
```

Comment:
Do we need `read` here as well?

Reply:
Kibana won't have read access. That way the secrets can't be surfaced in the UI.
Originally I wanted Kibana to have write only, and Enterprise Search to have read only. But because Enterprise Search has `manage`  I can't scope it the way I want without massive changes, so I decided having write for Enterprise Search as well is okay.
However I think Kibana shouldn't have read access.

Reply:
I see I had a typo in my PR about read/write access for Kibana, sorry if that caused confusion

Reply:
That makes sense, thanks!